### ISSUE TITLE
Remember the last expression that was used to create/update field 

### ISSUE THREAD

**Issue by baswein** [qgis/QGIS#36481]
**Feature description.**

In the spirit of why not ask for nice things to add to an amazing program. I have a number of fields that I regularly update in an attribute table. If when I opened the field calculator and selected update field QGIS would prefill the expression box with the last expression I used on that field it would save me having to figure out which of the recent or saved expression it might be. If I understand virtual fields this has the advantage of the field (but not the expression) being persistent outside of the QGIS file. It would also lead to the possibility of right clicking on the field header and prompting for an update. As well as the benefit of keeping a record in that QGIS file of the expression used to create that field.   

**Additional context**

**Comment by gioman** [comment]
@baswein the "recent" group is not enough?

**Comment by baswein** [comment]
Imagine a situation where 6 months ago I created a field with a formula. The recent expressions has a stack of expressions that may or may not be the one I used. In fact I am often trying a bunch of expressions so there is a list of very similar ones that quickly clog up the recent expressions. Am I using it wrong? In order to remember how the field was created I am either trying to bake it into the field name (why shapefile, why?) or saving a lot of expressions or taking a lot of notes. 

Are the recent expression enough. Sure it works and I love that they exist but in the spirit of transparency of data creation history and information always in the right place at the right time I think my proposal would be an improvement.    

"I saw a beggar leaning on his wooden crutch
He said to me, "you must not ask for so much"
And a pretty woman leaning in her darkened door
She cried to me, "hey, why not ask for more?""
Leonard Cohen - Bird On The Wire

**Comment by DelazJ** [comment]
@baswein Does the [user expression ](https://docs.qgis.org/testing/en/docs/user_manual/working_with_vector/expression.html#user-expressions-312)help (at least for the storing/finding part)?

**Comment by baswein** [comment]
Yes the user expression is very helpful and I am excited that it has been added but it requires a deliberate action on the users part with expectation that this expression might be needed in the future it also requires diligent naming to keep them clear and organized. The process that I am describing would instead be an automatic process that would make the expression available where and when it was needed whether or not you know you are going to need it in the future. 
Let me give you an example of where this would be helpful that I often encounter. I am working with one of my students and they are trying to figure out why some of their attribute numbers don't seem right. We look at there notes form last week and try to figure it out and usually end up creating a new field with a new expression that we puzzled out (if we are in QGIS we can look through the recent expressions in arcmap not so much). If instead we could open up the field calculator and have the specific expression that they used to create the field to see if it was right it would help (of course there are many other ways that they may have tangled things up). I don't imagine that the expressions take up a lot of memory and could be stored with the layer in the project file xml.

Imagine a virtual field except the only part that is virtual is the expression. It is available in that project for: Doublechecking/review, Refreshing the values, and tweaking the parameters.
